src/app/records/PresentationNodeLeaf.tsx

```diff
@@ -13,17 +15,103 @@ export default function PresentationNodeLeaf({
   ownedItemHashes,
   completedRecordsHidden,
   redactedRecordsRevealed,
+  sortRecordProgression,
 }: {
   node: DimPresentationNodeLeaf;
   ownedItemHashes?: Set<number>;
   completedRecordsHidden: boolean;
   redactedRecordsRevealed: boolean;
+  sortRecordProgression: boolean;
 }) {
+  const newNode = { ...node };
+  if (sortRecordProgression) {
+    /**
+     * RECORDS
+     */
+    if (node.records) {
+      newNode.records = orderBy(
+        newNode.records,
+        (record) => {
```

Comment:
Would be nice to extract the sort functions rather than have them inline.

Reply:
I made a function that takes a `DimPresentationNodeLeaf` and returns a shallow copy of that `DimPresentationNodeLeaf` that's sorted. Which file should I put the function in?

Reply:
It can be in the same file.